lib/src/rewrite.rs

```diff
@@ -654,17 +654,24 @@ pub fn move_commits(
                             [id.clone()].to_vec()
                         }
                     })
-                    // Exclude any of the new parents of the target commits, since we are
-                    // "inserting" the target commits in between the new parents and the new
-                    // children.
-                    .filter(|id| {
-                        !new_parent_ids
+                    // If the original parents of the new children are the new parents
+                    // of the `target_heads`, replace them with the target heads since
+                    // we are "inserting" the target commits in between the new parents and the
+                    // new children.
+                    .flat_map(|id| {
+                        if new_parent_ids
                             .iter()
-                            .any(|new_parent_id| new_parent_id == id)
+                            .any(|new_parent_id| *new_parent_id == id)
+                        {
+                            target_heads.clone()
+                        } else {
+                            [id.clone()].to_vec()
```

Comment:
nit: `vec![id]`

btw, this `.collect()`  would be better expressed as imperative `for` loop.